platforms/cuda/src/CudaTorchKernels.cpp

```diff
@@ -154,5 +157,7 @@ double CudaCalcTorchForceKernel::execute(ContextImpl& context, bool includeForce
         if (!outputsForces)
             posTensor.grad().zero_();
     }
+    cuCtxPopCurrent(&primary);
+    cuDevicePrimaryCtxRelease(cu.getDevice());
     return energyTensor.item<double>(); // This implicitly synchronize the PyTorch context
```

Comment:
The context is being used after it was popped

Reply:
As far as I can tell, simple operations on tensors don't cause any problems.  The only things that fail if we don't set the context first are `forward()` and `backward()`.